lean/main/elaboration.lean

```diff
@@ -318,11 +321,13 @@ def myanonImpl : TermElab := fun stx typ? => do
   let stx ← `($(mkIdent ctor) $args*) -- syntax quotations
   elabTerm stx typ -- call term elaboration recursively
 
-
 #check (⟨⟨1, sorry⟩⟩ : Fin 12) -- { val := 1, isLt := (_ : 1 < 12) } : Fin 12
 #check ⟨⟨1, sorry⟩⟩ -- type is not of the expected form ?_uniq.5991
 #check (⟨⟨0⟩⟩ : Nat) -- type doesn't have exactly one constructor
 
+-- I wanted to find an example for each type of error above.  I suggested one for
+-- "exactly one constructor", but could not find one for "expected type must be known"
```

Comment:
It will be very hard to find a situation where the type will not be a meta variable I think, the elaborator tends to use these quite a lot for its implicit argument stuff, I think I can refactor the code a little to make it work though.

Reply:
Well, it is not really important: it was just to shiw another behaviour, but I do not think that it is worth to make the code less natural just to include this failure.